openjdk25: init at 25+36

NyCodeGHG opened the pull request:
<!--
^ Please summarise the changes you have done and explain why they are necessary here ^

For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform:
  - [x] x86_64-linux
  - [x] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- Tested, as applicable:
  - [ ] [NixOS tests] in [nixos/tests].
  - [ ] [Package tests] at `passthru.tests`.
  - [ ] Tests in [lib/tests] or [pkgs/test] for functions and "core" functionality.
- [ ] Ran `nixpkgs-review` on this PR. See [nixpkgs-review usage].
- [x] Tested basic functionality of all binary files, usually in `./result/bin/`.
- Nixpkgs Release Notes
  - [ ] Package update: when the change is major or breaking.
- NixOS Release Notes
  - [ ] Module addition: when adding a new NixOS module.
  - [ ] Module update: when the change is significant.
- [x] Fits [CONTRIBUTING.md], [pkgs/README.md], [maintainers/README.md] and other READMEs.

[NixOS tests]: https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests
[Package tests]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests
[nixpkgs-review usage]: https://github.com/Mic92/nixpkgs-review#usage

[CONTRIBUTING.md]: https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md
[lib/tests]: https://github.com/NixOS/nixpkgs/blob/master/lib/tests
[maintainers/README.md]: https://github.com/NixOS/nixpkgs/blob/master/maintainers/README.md
[nixos/tests]: https://github.com/NixOS/nixpkgs/blob/master/nixos/tests
[pkgs/README.md]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md
[pkgs/test]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/test

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

NyCodeGHG committed:
openjdk25: init at 25+36

NyCodeGHG committed:
openjfx25: init at 25+29

`eclipse/updates/4.30/R-4.30-202312010110/plugins/org.eclipse.swt.gtk.linux.aarch64_3.124.200.v20231113-1355` has been manually added in deps.json,
as our gradle dependency lockfile generator currently cannot handle platform
specific dependencies like this one.

msgilligan commented:
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 446346`
Commit: `1a251700b73b9bf19d24782192a521ee3d0e94d6`

---
### `aarch64-darwin`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>jre25_minimal</li>
  </ul>
</details>

msgilligan commented on pkgs/by-name/op/openjfx/25/source.json:
```diff
@@ -0,0 +1,6 @@
+  "hash": "sha256-O8ogDcM/YYYwFmYqglc48m11lcbjqxeYbjsB6crY1bU=",
+  "owner": "openjdk",
+  "repo": "jfx25u",
+  "rev": "refs/tags/25+29"
```
Why is this not using https://github.com/openjdk/jfx25u/releases/tag/25-ga ?

msgilligan reviewed (changes requested):
I think jfx25u should be bumped to the latest, `-ga` version. There is also the `nixpkgs-vet` issue.

Oh, and nixpkgs-review failed on `aarch64-linux` for me. See below. (maybe the `-ga` version will fix it?)

Others than those things it looks good to me, but TBH I'm not familiar with the `openjdk` build.

I was also surprised to find that `jre25_minimal` was built on `aarch64-darwin`.  I'm assuming that was supposed to happen, but I had been under (the probably mistaken) assumption that `openjdk` was _always_ replaced by `zulu` on Darwin. I will investigate further (if for no other reason than to educate myself.)

msgilligan commented:
> I was also surprised to find that `jre25_minimal` was built on `aarch64-darwin`.

```
$ ./result/bin/java --version
openjdk 25 2025-09-16 LTS
OpenJDK Runtime Environment Zulu25.28+85-CA (build 25+36-LTS)
OpenJDK 64-Bit Server VM Zulu25.28+85-CA (build 25+36-LTS, mixed mode)
```

So it is Zulu, but for some reason only `jre25_minimal` is rebuilt by `nixpkgs-review`.

NyCodeGHG commented:
> There is also the nixpkgs-vet issue.

This is because this PR adds a new top-level attribute to all-packages.nix, but I don't see how we can avoid that without restructuring how JDKs are built in nixpkgs.

For the aarch64 build failure, we probably need something like https://github.com/NixOS/nixpkgs/pull/397129/commits/e527d18630173a7c54af821b2fe9ed315bd938e0 (and maybe figure out why this isn't included by default?)

msgilligan commented:
> > There is also the nixpkgs-vet issue.
> 
> This is because this PR adds a new top-level attribute to all-packages.nix, but I don't see how we can avoid that without restructuring how JDKs are built in nixpkgs.

@emilazy shared some tips with me that I used on the Zulu25 PR that might work here, see: https://github.com/NixOS/nixpkgs/pull/443517#issuecomment-3304438217

NyCodeGHG committed:
jre{11,17,21,25,}_minimal: group together in all-packages.nix

According to https://github.com/NixOS/nixpkgs/blob/5b751d1b50366c18fc5ef7faa82f5d4d2b75ef47/pkgs/by-name/README.md#recommendation-for-new-packages-with-multiple-versions

msgilligan reviewed (approved):
Looks great!  (I re-ran `nixpkgs-review` on both `aarch64-linux` and `aarch64-darwin` and it passes on both)

FliegendeWurst commented on pkgs/top-level/all-packages.nix:
```diff
@@ -5304,21 +5310,34 @@ with pkgs;
-  jre_minimal = callPackage ../development/compilers/openjdk/jre.nix {
-    jdkOnBuild = buildPackages.jdk;
-  };
+  inherit
```
I don't understand the reason for this refactor. Is the `inherit` version benefical in some way?

numinit commented:
java and jshell binaries tested and work. Thanks!
